src/sentry/integrations/typings/metric_alert_registry.py

```diff
@@ -44,3 +46,50 @@ def from_workflow_engine_models(
             detection_type=detector.config.get("detection_type"),
             comparison_delta=detector.config.get("comparison_delta"),
         )
+
+
+@dataclass
+class NotificationContext:
+    """
+    NotificationContext is a dataclass that represents the context required send a notification.
+    """
+
+    integration_id: int | None = None
+    target_identifier: str | None = None
+    target_display: str | None = None
+    sentry_app_config: list[dict[str, Any]] | dict[str, Any] | None = None
+    sentry_app_id: str | None = None
+
+    @classmethod
+    def from_alert_rule_trigger_action(cls, action: AlertRuleTriggerAction) -> NotificationContext:
```

Comment:
🤔 is there any overlap between this and the AlertContext that we could [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) up? should this also live in the `incidents` module since it's only for those notifications?